code/LeftAndMain.php

```diff
@@ -479,6 +484,23 @@ public function jsonError($errorCode, $errorMessage = null)
         throw new HTTPResponse_Exception($response);
     }
 
+    /**
+     * Check the CSRF token send in the request header in the context of making a JSON XHR request
+     */
+    protected function checkJsonCsrfToken(HTTPRequest $request): void
+    {
+        $securityToken = SecurityToken::inst();
+        $name = $securityToken->getName();
+        if ($name === 'SecurityID') {
+            // SecurityID is a non-standard HTTP header, it should be prefixed with 'X-'
+            $name = "X-SecurityID";
+        }
+        $value = $request->getHeader($name);
+        if (!$securityToken->check($value)) {
+            $this->jsonError(400);
+        }
```

Comment:
Why not just `SecurityToken::inst()->checkRequest($request)`? Seems like this whole method is unnecessary?

Reply:
That is correct, will update